src/module/item/condition/persistent-damage-editor.ts

```diff
@@ -1,30 +1,44 @@
 import type { ActorPF2e } from "@actor";
+import type { ApplicationConfiguration, ApplicationRenderContext } from "@client/applications/_module.mjs";
+import type { HandlebarsRenderOptions } from "@client/applications/api/_module.mjs";
 import { damageDiceIcon } from "@system/damage/helpers.ts";
 import { DamageRoll } from "@system/damage/roll.ts";
-import { DamageType } from "@system/damage/types.ts";
+import type { DamageType } from "@system/damage/types.ts";
 import { DAMAGE_TYPE_ICONS } from "@system/damage/values.ts";
 import { htmlClosest, htmlQuery, htmlQueryAll } from "@util";
 import * as R from "remeda";
 import type { PersistentDamagePF2e } from "./document.ts";
-import appv1 = foundry.appv1;
 
-class PersistentDamageDialog extends appv1.api.Application<PersistentDamageDialogOptions> {
-    constructor(
-        private actor: ActorPF2e,
-        options: Partial<PersistentDamageDialogOptions> = {},
-    ) {
-        super(options);
-        actor.apps[this.appId] = this;
-    }
-
-    static override get defaultOptions(): appv1.api.ApplicationV1Options {
-        return {
-            ...super.defaultOptions,
-            classes: ["persistent-damage-dialog"],
-            template: "systems/pf2e/templates/items/persistent-damage-dialog.hbs",
+class PersistentDamageEditor extends fa.api.HandlebarsApplicationMixin(fa.api.ApplicationV2) {
+    static override DEFAULT_OPTIONS: DeepPartial<fa.ApplicationConfiguration> = {
+        classes: ["persistent-damage-editor"],
+        position: {
             width: 380,
             height: "auto",
-        };
+        },
+        actions: {
+            add: PersistentDamageEditor.#onAdd,
+            delete: PersistentDamageEditor.#onDelete,
+            roll: PersistentDamageEditor.#onRoll,
+        },
+    };
+
+    static override PARTS: Record<string, fa.api.HandlebarsTemplatePart> = {
+        main: {
+            template: "systems/pf2e/templates/items/persistent-damage-editor.hbs",
+            root: true,
+        },
+    };
+
+    actor: ActorPF2e;
+
+    editing?: string;
```

Comment:
```suggestion
    editing: string | null = null;
```
The property is always present (assigned by the constructor).